var/spack/repos/builtin/packages/py-isort/package.py

```diff
@@ -23,7 +23,8 @@ class PyIsort(PythonPackage):
 
     depends_on("python@3.8:", when="@5.12:", type=("build", "run"))
     depends_on("python@3.6.1:3", when="@5:5.10", type=("build", "run"))
-    depends_on("py-poetry-core@1:", type="build")
+    depends_on("py-setuptools", when="@:4", type="build")
```

Comment:
```suggestion
    depends_on("py-setuptools", when="@:4", type=('build', 'run'))
```
`py-setuptools` is also used during runtime for versions `@:4`, see [isort/main.py#L29 vor version 4.3.20](https://github.com/PyCQA/isort/blob/4.3.20/isort/main.py#L29)

Remark: The `py-setuptools` dependency was removed in #27798

Reply:
It's still an optional dependency in `@5:`, maybe best to add it unconditionally? Optional means it just tries to import it, so at runtime it may pick up a random user setuptools package.

Reply:
Usually we don't add optional dependencies without a variant. So I would tend to not add it if not necessary because of any other reason.

Reply:
setuptools is a universal dependency and given isort imports it unconditionally, I don't see any downside of requiring the optional dep

Reply:
I don't think there is a downside for the package except of having `py-setuptools` as run dependency again which @adamjstewart most of the times tries to avoid if not necessary (as far as I understand). Maybe let him decide on this since he was the one who removed it?

Reply:
I agree with @manuelakuhn, let's leave the dependency off unless you want to add a variant.